pulsar-common/src/main/java/org/apache/pulsar/common/util/RateLimiter.java

```diff
@@ -203,7 +203,7 @@ public synchronized boolean tryAcquire(long acquirePermit) {
      *
      * @return returns 0 if permits is not available
      */
-    public synchronized long getAvailablePermits() {
+    public long getAvailablePermits() {
```

Comment:
There are a lot of methods using `synchronized` in RateLimiter currently. Does this change address an actual problem?
The design of RateLimiter isn't optimal, but changing this method makes the implementation inconsistent. I think it would be better to first clarify the issue with using syncronization and redesign a RateLimiter implementation with minimal use of blocking methods and synchronization.

Reply:
The synchronized keyword blocked the RateLimiters to simultaneously access each other's available permits. Since each RateLimiter is a side-task that runs it's own instance of this function. 

Since this function only reads from memory, and the variables which are read are `volatile` means it'll be read directly from the main memory, I believe it shouldn't damage anything.

I'm still pretty new to Java Concurrency though.

Reply:
it's not damaging, but it's not needed for this fix. It's better to keep the fix for the actual problem minimal and propose a separate change for improving the design and implementation of RateLimiter.

Reply:
This change is necessary to prevent a dead lock when two rate limiters will attempt to use `getAvailablePermits()` and is required for the fix.

Reply:
Makes sense. Thanks for clarifying.